db/memtable_list.cc

```diff
@@ -438,7 +438,6 @@ void MemTableList::RollbackMemtableFlush(const autovector<MemTable*>& mems,
   TEST_SYNC_POINT("RollbackMemtableFlush");
```

Comment:
Just want to understand more which scenario this is catering for, when will this `RollbackMemtableFlush` be called with empty memtables?

Reply:
One possible scenario is that 
- switch memtable -> schedules flush 1
- switch memtable -> schedules flush 2
- then flush 1 starts and pick both memtables
- flush 2 will not pick any memtable

Reply:
I see, I was thinking why this assertion always hold true previously for the `RollbackMemtableFlush()` calls, the scenario you described would happen then too, right?

Reply:
Good point. I am not sure why it did not manifest before. From code here 

https://github.com/facebook/rocksdb/blob/6afde142663aca7081daf44fa01484fe92e70e13/db/db_impl/db_impl_compaction_flush.cc#L2877-L2889

the scenario I mentioned above should only happen when atomic_flush=true. Maybe we don't have a lot of tests with atomic_flush.